### PR TITLE
vim-patch:8.1.1259: crash when exiting early

### LINKED ISSUE #4334: V8.1.1253: Core Dump on missing parameter argument
Vim 8.1.1253 crashes with sigsegv when specific command line parameter are missing a required argument.

Reproducible with:
- `vim --servername`
- `vim --display`

Both parameter are handled in `early_arg_scan(...)`.

OS: Linux x86_64

### PR DESCRIPTION
**Pull request by VVKot**
Problem:    Crash when exiting early. (Ralf Schandl)
Solution:   Only pop/push the title when it was set. (closes vim/vim#4334)
https://github.com/vim/vim/commit/e5c83286bb9a72cc686f2826e605eddebe3c730c

### CODE REVIEW

**Comment by VVKot** [comment]
@janlazo I think test failures here are flakes?

**Comment by janlazo** [comment]
probably but ASAN build shouldn't fail.

**Comment by VVKot** [comment]
@janlazo ASAN good to go

**Comment by VVKot** [comment]
@janlazo should we merge this?

**Comment by zeertzjq** [comment]
I think all the `NULL` checks here are always true because `win_alloc_first()` is always called from `early_init()` before `command_line_scan()`.

File: src/nvim/memory.c [hunk 1]
```diff
@@ -594,19 +594,23 @@ void free_all_mem(void)
...
-  // Clear mappings, abbreviations, breakpoints.
-  do_cmdline_cmd("lmapclear");
-  do_cmdline_cmd("xmapclear");
-  do_cmdline_cmd("mapclear");
-  do_cmdline_cmd("mapclear!");
-  do_cmdline_cmd("abclear");
-  do_cmdline_cmd("breakdel *");
-  do_cmdline_cmd("profdel *");
-  do_cmdline_cmd("set keymap=");
+  // When exiting from mainerr_arg_missing curbuf has not been initialized,
```
Comment (zeertzjq) [inline src/nvim/memory.c:597]: There is also no `mainerr_arg_missing` function.

**Comment by zeertzjq** [comment]
Perhaps this should be N/A?

**Pull request closed without merging**